module/zfs/zfs_ioctl.c

```diff
@@ -7562,11 +7574,11 @@ zfsdev_ioctl_common(uint_t vecnum, zfs_cmd_t *zc, int flag)
 		char *s = tsd_get(zfs_allow_log_key);
 		if (s != NULL)
 			kmem_strfree(s);
-		(void) tsd_set(zfs_allow_log_key, saved_poolname);
-	} else {
-		if (saved_poolname != NULL)
-			kmem_strfree(saved_poolname);
+		(void) tsd_set(zfs_allow_log_key, kmem_strdup(saved_poolname));
 	}
+	if (saved_poolname != NULL)
+		kmem_free(saved_poolname, saved_poolname_len);
```

Comment:
Since we have to support `kmem_strfree(saved_poolname)` for `zfs_ioc_log_history()`, maybe we should use it here too?

Reply:
This string `saved_poolname` is truncated, so we can not call `kmem_strfree()`.  Either we use kmem_free( .., orig_size); or we restore the string (changing the nul-byte we poked in to something else)

Reply:
aaaaah.  Reading comprehension fail on my part, please disregard.